### DISCUSSION TITLE
How to prevent conflict with Livewire component attributes?

### DISCUSSION THREAD

**Discussion by mikebronner** [alpinejs/alpine#1309] [1. Help] [answered]
I'm probably missing something, but I couldn't find any documentation for this. I have a Livewire component within an Alpine `x-data` block, and Alpine keeps throwing errors for instance attributes, thinking that it is using `x-bind` shorthand, when it  should not be.

Here's an example:

```php
    <div
        x-data="{}"
    >
        <livewire:mycomponent
            :name="$name"
        />
    </div>
```

Alpine throws a client-side error:
```
Alpine Error: "ReferenceError: Can't find variable: $name"
```

I'm sure I'm missing something incredibly simple. Thanks for any help!

**Comment by ryangjchandler** [comment, marked as answer]
@mikebronner This shouldn't be happening - Livewire should be compiling the `<livewire:` tags on the server and removing them from the final HTML sent to the client. I'd suggest opening an issue on the Livewire repo.

**Reply by mikebronner** [reply]
@ryangjchandler doh! Yea, I should have thought of that. Seems like Livewire isn't compiling server-side then. I'll dig into my code, I'm sure it's something on my end. Thanks!

**Reply by ryangjchandler** [reply]
> @ryangjchandler doh! Yea, I should have thought of that. Seems like Livewire isn't compiling server-side then. I'll dig into my code, I'm sure it's something on my end. Thanks!

My best guess is that you've got the attributes on a new line - perhaps the RegEx that Livewire is using doesn't support that.

**Reply by mikebronner** [reply]
I doubt it, I have them on new lines all the time. I think it's an issue with me not registering the component correctly.

**Reply by ryangjchandler** [reply]
Ah, in which case Livewire should be throwing an exception right?

**Reply by mikebronner** [reply]
It ended up being the registration of the Livewire component in my package had a typo. Thanks for bearing with me. :)
